cli/src/main/java/com/devonfw/tools/ide/tool/ToolCommandlet.java

```diff
@@ -404,4 +449,49 @@ public void setVersion(VersionIdentifier version, boolean hint) {
     }
   }
 
+  /**
+   * Sets the tool edition in the environment variable configuration file.
+   *
+   * @param edition the edition to set.
+   */
+  public void setEdition(String edition) {
+
+    if ((edition == null) || edition.isBlank()) {
+      throw new IllegalStateException("Edition has to be specified!");
+    }
+
+    if (!Files.exists(this.context.getUrls().getEdition(getName(), edition).getPath())) {
+      this.context.warning("Edition {} seems to be invalid", edition);
+
+    }
```

Comment:
shouldn't these check go to line #477 (`setEdition(String, boolean)`)?

Reply:
Done